File: opspilot-enterprise/packages/shared-schema/src/opspilot_schema/notification.py
from __future__ import annotations
from typing import Optional
from pydantic import BaseModel


class NotificationItem(BaseModel):
    id: str
    title: str
    content: str
    priority: str
    status: str
    incident_ref: Optional[str] = None
    channels: list[str] = []
    recipients: list[str] = []
    created_at: str
    delivered_at: Optional[str] = None
    acknowledged_at: Optional[str] = None
    acknowledged_by: Optional[str] = None
    escalation_count: int = 0
    next_escalation_at: Optional[str] = None


class OnCallShift(BaseModel):
    id: str
    name: str
    team: str
    members: list[str]
    start_at: str
    end_at: str
    active: bool

File: packages/shared-types/src/notification.ts
export type NotificationStatus = "pending" | "delivered" | "acknowledged" | "escalated" | "timed_out";
export type NotificationChannel = "dingtalk" | "wecom" | "email" | "sms" | "webhook" | "phone";
export type NotificationPriority = "low" | "normal" | "high" | "urgent";

export interface NotificationItem {
  id: string;
  title: string;
  content: string;
  priority: NotificationPriority;
  status: NotificationStatus;
  incident_ref: string | null;
  channels: NotificationChannel[];
  recipients: string[];
  created_at: string;
  delivered_at: string | null;
  acknowledged_at: string | null;
  acknowledged_by: string | null;
  escalation_count: number;
  next_escalation_at: string | null;
}

export interface OnCallShift {
  id: string;
  name: string;
  team: string;
  members: string[];
  start_at: string;
  end_at: string;
  active: boolean;
}
